### DISCUSSION TITLE
In 5.3 video 2 how is train_acc callable like a function

### DISCUSSION THREAD

**Discussion by bonosa** [Lightning-AI/dl-fundamentals#25] [Lecture videos and quizzes] [answered]
if it's an attribute?

https://github.com/Lightning-AI/dl-fundamentals/blob/main/unit05-lightning/5.3-torchmetrics/5.3-part2-lightning-torchmetrics.py

chatGPT says it's not a regular attribute but an instance of the Accuracy class from torchmetrics. Is that correct?

**Comment by rasbt** [comment, marked as answer]
It's an object we instantiate from the `Accuracy` class, and then we assign it to an attribute of the `LightningModel`. The `train_acc` object then has a `.forward()` method that is executed when we call `train_acc(...)`. 
I have a blog article that might be helpful to understand what's happening: https://sebastianraschka.com/blog/2022/torchmetrics.html

**Reply by bonosa** [reply]
the article cleared my doubts. Thanks!
